tests/common/dualtor/tor_failure_utils.py

```diff
@@ -0,0 +1,124 @@
+"""
+Tor Failure utilities to test switchover and MUX handling during:
+Shutdown all BGP sessions on a ToR
+Shutdown the LinkProber on a ToR
+Blackhole all traffic on a ToR
+Reboot a ToR
+"""
+from tests.common.reboot import *
+import ipaddress
+import pytest
+import logging
+import time
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture
+def shutdown_tor_bgp():
+    """
+    Shutdown all BGP sessions 
+    """
+    torhost = []
+    def shutdown_tor_bgp(duthost, shutdown_all=True):
+        torhost.append(duthost)
+        bgp_neighbors = duthost.get_bgp_neighbors()
+        up_bgp_neighbors = [ k.lower() for k, v in bgp_neighbors.items() if v["state"] == "established" ]
+        if shutdown_all and up_bgp_neighbors:
+            logger.info("Shutdown BGP sessions on {}".format(duthost.hostname))
+            duthost.shell("config bgp shutdown all")
+
+    yield shutdown_tor_bgp
+
+    time.sleep(1)
+    if torhost:
```

Comment:
Why is this necessary? Are we expecting `duthost` to become undefined at some point during the test?